pkg/lint/lintersdb/manager.go

```diff
@@ -1,931 +1,131 @@
 package lintersdb
 
 import (
-	"regexp"
+	"os"
+	"sort"
+
+	"golang.org/x/exp/maps"
 
 	"github.com/golangci/golangci-lint/pkg/config"
-	"github.com/golangci/golangci-lint/pkg/golinters"
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
 	"github.com/golangci/golangci-lint/pkg/lint/linter"
 	"github.com/golangci/golangci-lint/pkg/logutils"
 )
 
+// EnvTestRun value: "1"
+const EnvTestRun = "GL_TEST_RUN"
+
+type Builder interface {
+	Build(cfg *config.Config) []*linter.Config
+}
+
 type Manager struct {
+	log    logutils.Log
+	debugf logutils.DebugFunc
+
 	cfg *config.Config
-	log logutils.Log
 
-	nameToLCs     map[string][]*linter.Config
-	customLinters []*linter.Config
+	linters []*linter.Config
+
+	nameToLCs map[string][]*linter.Config
 }
 
-func NewManager(cfg *config.Config, log logutils.Log) *Manager {
-	m := &Manager{cfg: cfg, log: log}
-	m.customLinters = m.getCustomLinterConfigs()
+func NewManager(log logutils.Log, cfg *config.Config, builders ...Builder) (*Manager, error) {
```

Comment:
I don't like to use a constructor that calls other constructors (that build structures with a state), this creates a hard dependency, which can be a problem for testing.

Also, I want to keep the variadic because I have an idea for a new plugin system that will use this.

Reply:
I will add godoc.

Reply:
Yes, sounds very reasonable and I agree! I was more curious about future plans. To me this is some gray area because we never need a manager without linters but it's a low cost to pass it like you do.

Reply:
I added some godoc, if you want to check them.

Reply:
> I was more curious about future plans.

After this PR, I will open 3 PRs (2 proposals) at the same time (because this time they are not linked), and we will be able to discuss that together.
I hope you will appreciate my proposals.

Reply:
LGTM!